### DISCUSSION TITLE
How to reset password directly via database?

### DISCUSSION THREAD

**Discussion by nandgator** [firefly-iii/firefly-iii#6062] [Firefly III]
I do not have a SMTP setup. So there's no way to rest password. But I've direct access to the (mariadb) database. In the `user` table, I see:

```
+------------------------+--------------------------------------------------------------+
| email                  | password                                                     |
+------------------------+--------------------------------------------------------------+
| my_email@ddres.net     | some_large_hash                                              |
+------------------------+--------------------------------------------------------------+
```

I tried setting the password to blank (`""`). But I cannot login with an empty password.

How can I reverse that hash to get the password?

**Comment by JC5** [comment]
The email messages end up in your log files, so you can read them from there.
